applications/nightly_build/test_ssrnet.py

```diff
@@ -0,0 +1,232 @@
+        pred_a_s2 = Reshape((self.num_classes, self.stage_num[1]))(feat_pred_s2)
+        # -------------------------------------------------------------------------------------------------------------------------
+        s_layer2 = Conv2D(64, (1, 1), activation='tanh')(s_layer2)
+        s_layer2 = MaxPooling2D((2, 2))(s_layer2)
+
+        x_layer2 = Conv2D(64, (1, 1), activation='relu')(x_layer2)
+        x_layer2 = AveragePooling2D((2, 2))(x_layer2)
+
+        feat_s3_pre = Multiply()([s_layer2, x_layer2])
+        feat_s3_pre = Flatten()(feat_s3_pre)
+        feat_delta_s3 = Dense(2 * self.num_classes, activation='tanh')(feat_s3_pre)
+        delta_s3 = Dense(self.num_classes, activation='tanh', name='delta_s3')(feat_delta_s3)
+
+        feat_local_s3 = Dense(2 * self.num_classes, activation='tanh')(feat_s3_pre)
+        local_s3 = Dense(units=self.num_classes, activation='tanh', name='local_delta_stage3')(feat_local_s3)
+
+        feat_pred_s3 = Dense(self.stage_num[2] * self.num_classes, activation='relu')(feat_s3_pre)
+        pred_a_s3 = Reshape((self.num_classes, self.stage_num[2]))(feat_pred_s3)
+
+        # -------------------------------------------------------------------------------------------------------------------------
+
+        def SSR_module(x, s1, s2, s3, lambda_d):
+            a = x[0][:, :, 0] * 0
+            b = x[0][:, :, 0] * 0
+            c = x[0][:, :, 0] * 0
+
+            di = s1 // 2
+            dj = s2 // 2
+            dk = s3 // 2
+
+            V = 99
+            # lambda_d = 0.9
+
+            for i in range(0, s1):
+                a = a + (i - di + x[6]) * x[0][:, :, i]
+            # a = K.expand_dims(a,-1)
+            a = a / (s1 * (1 + lambda_d * x[3]))
+
+            for j in range(0, s2):
+                b = b + (j - dj + x[7]) * x[1][:, :, j]
+            # b = K.expand_dims(b,-1)
+            b = b / (s1 * (1 + lambda_d * x[3])) / (s2 * (1 + lambda_d * x[4]))
+
+            for k in range(0, s3):
+                c = c + (k - dk + x[8]) * x[2][:, :, k]
+            # c = K.expand_dims(c,-1)
+            c = c / (s1 * (1 + lambda_d * x[3])) / (s2 * (1 + lambda_d * x[4])) / (s3 * (1 + lambda_d * x[5]))
+
+            pred = (a + b + c) * V
+
+            return pred
+
+        pred_pose = Lambda(SSR_module,
+                           arguments={'s1': self.stage_num[0], 's2': self.stage_num[1], 's3': self.stage_num[2],
+                                      'lambda_d': self.lambda_d}, name='pred_pose')(
+            [pred_a_s1, pred_a_s2, pred_a_s3, delta_s1, delta_s2, delta_s3, local_s1, local_s2, local_s3])
+
+        model = Model(inputs=img_inputs, outputs=pred_pose)
+
+        return model
+
+
+class TestSSRNet(unittest.TestCase):
+
+    def setUp(self):
+        self.model_files = []
+
+    def tearDown(self):
+        for fl in self.model_files:
+            os.remove(fl)
+
+    @unittest.skipIf(not is_weekend,
```

Comment:
what do you think the weekend is different from weekday, regarding to running the tests?

Reply:
On weekdays, we sometimes need test nightly build for import PR changes, we don't want to take too long but cover important models. On weekend, we can test even more (like these models) and spend more time.

Reply:
try the pipeline variable for the manual check:
https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=azure-devops&tabs=yaml%2Cbatch